crates/backend/src/opendal.rs

```diff
@@ -34,6 +35,30 @@ fn runtime() -> &'static Runtime {
     })
 }
 
+/// Throttling parameters
+///
+/// Note: Throttle implements [`FromStr`] to read it from something like "10kiB,10MB"
+#[derive(Debug, Clone, Copy)]
+pub struct Throttle {
+    bandwidth: u32,
+    burst: u32,
+}
+
+impl FromStr for Throttle {
+    type Err = Error;
+    fn from_str(s: &str) -> Result<Self> {
+        let mut values = s
+            .split(',')
+            .map(|s| ByteSize::from_str(s).map_err(|err| anyhow!("Error: {err}")))
+            .map(|b| -> Result<u32> { Ok(b?.as_u64().try_into()?) });
+        let bandwidth = values
+            .next()
+            .ok_or_else(|| anyhow!("no bandwidth given"))??;
+        let burst = values.next().ok_or_else(|| anyhow!("no burst given"))??;
+        Ok(Self { bandwidth, burst })
+    }
+}
+
```

Comment:
Can we implement a unit test for this, please? I think it makes sense to have that whenever we parse values from user input, to make sure it works as expected. Also, we add new code. The test coverage doesn't throw a warning here, because I took rustic_backend out of the reporting, due to complexity to test foreign code, but if we work with parsing and this may actually affect the user, it would be good, if this has test cases.

Reply:
unit tests are added.